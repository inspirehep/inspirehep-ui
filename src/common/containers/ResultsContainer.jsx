import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Immutable from 'immutable';

class ResultsContainer extends Component {
  render() {
    return (
      <div>
        {this.props.results.map(result => (
          <div className="mv3" key={result.get('id')}>
            {this.props.renderItem(result)}
          </div>
        ))}
      </div>
    );
  }
}

ResultsContainer.propTypes = {
  results: PropTypes.instanceOf(Immutable.List),
  renderItem: PropTypes.func.isRequired,
};

ResultsContainer.defaultProps = {
  results: Immutable.List(),
};

const stateToProps = state => ({
  results: state.search.get('results'),
});

export default connect(stateToProps)(ResultsContainer);
