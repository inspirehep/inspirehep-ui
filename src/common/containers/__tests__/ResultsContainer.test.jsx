import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import { getStoreWithState } from '../../../fixtures/store';
import ResultsContainer from '../ResultsContainer';

describe('ResultsContainer', () => {
  it('renders initial state ', () => {
    const store = getStoreWithState({
      search: fromJS({
        results: [
          {
            id: 1,
            value: 'value1',
          },
          {
            id: 2,
            value: 'value2',
          },
        ],
      }),
    });
    const wrapper = shallow(
      <ResultsContainer
        store={store}
        renderItem={result => <span>{result.get('value')}</span>}
      />
    ).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
