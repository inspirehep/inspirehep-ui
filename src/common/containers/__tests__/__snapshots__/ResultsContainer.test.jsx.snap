// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsContainer renders initial state  1`] = `
<div>
  <div
    className="mv3"
    key="1"
  >
    <span>
      value1
    </span>
  </div>
  <div
    className="mv3"
    key="2"
  >
    <span>
      value2
    </span>
  </div>
</div>
`;
